test/api/endpoints/bfLoss/amend/v3/connector/AmendBFLossConnectorSpec.scala

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package api.endpoints.bfLoss.amend.v3.connector
+
+import api.connectors.ConnectorSpec
+import api.endpoints.bfLoss.amend.anyVersion.request.AmendBFLossRequestBody
+import api.endpoints.bfLoss.amend.v3.request.AmendBFLossRequest
+import api.endpoints.bfLoss.amend.v3.response.AmendBFLossResponse
+import api.endpoints.bfLoss.connector.v3.BFLossConnector
+import api.endpoints.bfLoss.domain.v3.TypeOfLoss
+import api.models.errors.{ LossIdFormatError, MultipleErrors, NinoFormatError, SingleError }
+import api.models.ResponseWrapper
+import api.models.domain.Nino
+
+import scala.concurrent.Future
+
+class AmendBFLossConnectorSpec extends ConnectorSpec {
+
+  val nino: String   = "AA123456A"
+  val lossId: String = "AAZZ1234567890a"
+
+  val requestBody: AmendBFLossRequestBody = AmendBFLossRequestBody(500.13)
+  val request: AmendBFLossRequest         = AmendBFLossRequest(nino = Nino(nino), lossId = lossId, requestBody)
+
+  trait Test {
+    _: ConnectorTest =>
+
+    val connector: BFLossConnector = new BFLossConnector(http = mockHttpClient, appConfig = mockAppConfig)
+
+  }
+
+  "amendBFLosses" when {
+    "downstream returns OK" should {
+
+      val response: AmendBFLossResponse = AmendBFLossResponse(
+        businessId = "XKIS00000000988",
+        typeOfLoss = TypeOfLoss.`self-employment`,
+        lossAmount = 500.13,
+        taxYearBroughtForwardFrom = "2019-20",
+        lastModified = "2018-07-13T12:13:48.763Z"
+      )
+
+      "a valid non-TYS request with no provided parameters is supplied" in new IfsTest with Test {
```

Comment:
the scenario is a bit ambiguous, "no provided parameters" but it does have path parameters. Also, active voice scenario text always reads better! (and tends towards being more specific) - maybe:

"the connector sends a non-TYS request with no query params"

Reply:
but tbh the `should` text should probably be an actual "should".

The two tests (success & failure) could actually be grouped, e.g.
```
"amendBFLosses" should {
  "return the expected response" when {
    "downstream returns OK" in new IfsTest with Test {
    }
    "downstream returns an error" in new IfsTest with Test {
    }
  }
}
```

Reply:
(same comment would apply to the other tests in the PR)